liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -299,44 +288,169 @@ public ServerTask getServerTask() throws Exception {
+            if (p != null) {
+                Enumeration<?> e = p.propertyNames();
+                while (e.hasMoreElements()) {
+                    String key = (String) e.nextElement();
+                    if (key.startsWith(prefix)) {
+                        result.put(key, p.get(key));
+                    }
+                }
+            }
+            return result;
+        }
+
+        private List<Dependency> getEsaDependency(List<Dependency> dependencies) {
+            List<Dependency> deps = new ArrayList<Dependency>();
+            if (dependencies != null) {
+                for (Dependency d : dependencies) {
+                    if ("esa".equals(d.getType())) {
+                        deps.add(d);
+                    }
+                }
+            }
+            return deps;
+        }
+        private List<Dependency> getCompileDependency(List<Dependency> dependencies) {
+            List<Dependency> deps = new ArrayList<Dependency>();
+            if (dependencies != null) {
+                for (Dependency d : dependencies) {
+                    if ("compile".equals(d.getScope())) {
+                        deps.add(d);
+                    }
+                }
+            }
+            return deps;
+        }
+
+        private boolean hasServerPropertyChanged(MavenProject project, MavenProject backupProject) {
+            Properties p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.bootstrap.");
+            Properties q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.bootstrap.");
+            if (!Objects.equals(p, q)) {
+                return true;
+            } else {
+                p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.jvm.");
+                q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.jvm.");
+                if (!Objects.equals(p, q)) {
+                    return true;
+                } else {
+                    p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.env.");
+                    q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.env.");
+                    if (!Objects.equals(p, q)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        private boolean hasServerVariableChanged(MavenProject project, MavenProject backupProject) {
+            Properties p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.var.");
+            Properties q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.var.");
+            if (!Objects.equals(p, q)) {
+                return true;
+            } else {
+                p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.defaultVar.");
+                q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.defaultVar.");
+                if (!Objects.equals(p, q)) {
```

Comment:
same comment about getProperties and else statements

Reply:
ok